src/Microsoft.AspNetCore.DataProtection/NullLoggerFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.DataProtection
+{
+    /// <summary>
+    /// An <see cref="ILoggerFactory"/> that provides a logger that no ops.
+    /// </summary>
+    public class NullLoggerFactory : ILoggerFactory
```

Comment:
Yeah I think we should consolidate all these `NullLogger***` types in one place. Can you log a bug in the Logging repo so we can clean this up, then consume it from here?